hack/codegen.sh

```diff
@@ -19,54 +19,38 @@ pricing() {
 }
 
 skugen() {
-   SUBSCRIPTION_ID=$(az account show --query 'id' --output tsv)
-  export SUBSCRIPTION_ID
-  if [ -z "${SUBSCRIPTION_ID}" ]; then
-    echo "No subscription is set. Please login and set a subscription."
-    return 1
+  location=${1:-eastus}
```

Comment:
what does this do? does it basically say "use args[1] or default to eastus"? not sure i understand :)

Reply:
Yes, exactly. This is just idiomatic bash.

Explaining something like this is where Copilots shine, BTW. Use your favorite integration, or just from directly from GitHub:

<img width="925" alt="image" src="https://github.com/Azure/karpenter/assets/18728999/aeefd619-bdbe-49f4-bcc0-9909f78e691c">

You will get this:

<img width="629" alt="image" src="https://github.com/Azure/karpenter/assets/18728999/637de100-fda4-4673-afbb-a4b92729cb87">

Reply:
that's awesome